### DISCUSSION TITLE
Is there a way to install queries?

### DISCUSSION THREAD

**Discussion by tsuza** [nvim-treesitter/nvim-treesitter#5608] [Q&A] [answered]
You can easily add and install a parser, which is great. However, how about downloading queries? It'd be necessary for languages that are not present by default in `nvim-treesitter`, without forcing the end-user to manually get the query files and dump them into a folder.

**Comment by clason** [comment]
No. Queries are editor specific, and not all parser repos contain them. And "end-users" don't have to do anything because nvim-treesitter takes care of it -- that's the whole point of this plugin. 

(If you're missing queries, you haven't installed the plugin correctly. Don't lazy-load unless you know exactly what you're doing.)

**Reply by tsuza** [reply]
Thanks for your time. There was a typo in my original message. I meant "...languages that are *not present by default...".

**Reply by clason** [reply]
I don't understand. If they're not in nvim-treesitter by default, where would you expect to get the queries from?

**Reply by tsuza** [reply]
From a github repository, for example. There are languages that are not present by default ( Sourcepawn for example, which is a fork of Pawn, based on small-c ). There are some projects that have language-specific parsers and queries for it.

**Reply by clason** [reply]
_which_ github repository? again, most parser repositories do _not_ contain queries, and those that do are not compatible with Neovim. With very few exceptions, all queries in this repo are custom-written or significantly adapted for use in Neovim.

**Reply by clason** [reply]
And we welcome PRs adding new parsers and (compatible) queries.

**Reply by tsuza** [reply]
[Here.](https://github.com/nilshelmig/tree-sitter-sourcepawn) It'll probably get PR'ed once it's fully ready, but until then it's a manual installation ( the queries, that is ).

( Edit: I've edited the original link with another repository, which is supposed to be more up-to-date than the other one. )

**Reply by clason** [reply]
Yeah, those queries would not be acceptable as-is. But that can be discussed in a PR.

**Reply by tsuza** [reply]
That's a story for another day. Though, if you don't mind, could you say what you didn't like ( as in, bad practices, things that could have been done differently ) that you noticed on a first glance while taking a look? It's so I can either try to PR the original repository with some improvements or at least make an issue there so someone that has more knowledge about this can take a look.

Anyway, back to the original question, I take it that there are no plans to make it easier to install third party queries for languages that don't already have it by default?

**Reply by clason** [reply, marked as answer]
Wrong capture names, wrong predicates. Please read CONTRIBUTING.md and  `:h vim.treesitter`. (Again, queries are editor-specific; upstream likely won't care.)

> Anyway, back to the original question, I take it that there are no plans to make it easier to install third party queries for languages that don't already have it by default?

Yes, there are no plans for this since it's not possible/feasible in general. If you don't want to use the curated queries from nvim-treesitter, just copy them yourself; you don't need this plugin then.

**Reply by amaanq** [reply]
>  I take it that there are no plans to make it easier to install third party queries for languages that don't already have it by default?

No, queries are not editor-agnostic nor will they always conform to the neovim precedence so to install them from an upstream repo blindly makes no sense

**Reply by tsuza** [reply]
Got it. Thank you all for your time and answers. Have a wonderful day.
